torch/testing/_internal/opinfo/core.py

```diff
@@ -1167,6 +1167,26 @@ def error_inputs(self, device, **kwargs):
         """
         return self.error_inputs_func(self, device, **kwargs)
 
+    def sample_inputs_sparse(
+        self, layout, device, dtype, requires_grad=False, **kwargs
+    ):
+        """Returns an iterable of SampleInputs that contain inputs with a
+        specified sparse layout.
+        """
+        if layout is torch.sparse_coo:
```

Comment:
Just a python nit, I think you could create a dictionary here
```
{
torch.sparse_coo: self.sample_inputs_sparse_coo,
torch. sparse_csr: self. sample_inputs_sparse_csr,
[...],
}
```

Or (I think) you could generate it with 
```
{getattr(torch, 'sparse_' + name): getattr(self, 'sample_inputs_sparse_' + name) for name in ['coo', 'csr', 'csc', 'bsr', 'bsc']}
```

Reply:
Right, in fact, there exist many other ways to make it a one-line, say:
```python
sample_inputs_mth = getattr(self, f'sample_inputs_{str(layout)[6:]}')
```
Would that be better?

Reply:
@pearu - Up to you. Just pointing it out.

Reply:
Now using:
```python
sample_inputs_mth = getattr(self, 'sample_inputs_' + str(layout).split('.', 1)[-1])
```